bot/extensions/language_cog.py

```diff
@@ -70,10 +74,39 @@ async def penguin_react(self, message: Message):
             if not fail:
                 await message.add_reaction(linus_trigger.positive_emoji)
 
+    async def pun_react(self, message: Message):
+        message_tokens = self.tokenizer.tokenize(message.content)
+        tokenlist = set(map(lambda s: s.lower(), message_tokens))
+
+        pun_words = PunWord.all()
+        word_set = set([pun_word.word for pun_word in pun_words])
+
+        matches = tokenlist.intersection(word_set)
+
+        if len(matches) > 0:
+            matched_pun_words = [
+                pun_word for pun_word in pun_words if pun_word.word in matches]
+
+            puns = set([Pun.get(pun_word.pun_id)
```

Comment:
It's not hurting anything if it's says a set so I'm not gonna make a big case about it but I'm pretty sure it doesn't need to since match_pun_words is created by 2 sets and, thus, it should already be unique.